concepts/security/README.md

```diff
@@ -33,28 +34,48 @@ Dapr also supports strong identities when deployed on Kubernetes, relying on a p
 By default, a workload cert is valid for 24 hours and the clock skew is set to 15 minutes.
 
 Mutual TLS can be turned off/on by editing the default configuration that is deployed with Dapr via the `spec.mtls.enabled` field.
-This can be done for both Kubernetes and self hosted modes.
-Specific details for how to do that can be found [here](../../howto/configure-mtls).
+This can be done for both Kubernetes and self hosted modes. Details for how to do this can be found [here](../../howto/configure-mtls).
+
+### mTLS self hosted
+The diagram below shows how the Sentry system service issues certificates for applications based on the root/issuer certificate that is provided by an operator or generated by the Sentry service as stored in a file
+
+<a href="https://ibb.co/Twc7P82"><img src="https://i.ibb.co/x2hBzjL/Screen-Shot-2020-03-30-at-6-21-54-PM.png" alt="Screen-Shot-2020-03-30-at-6-21-54-PM" border="0"></a>
 
 ### mTLS in Kubernetes
+The diagram below shows how the Sentry system service issues certificates for applications based on the root/issuer certificate that is provided by an operator or generated by the Sentry service as stored as a Kubernetes secret
 
-<a href="https://ibb.co/4VXWJ3d"><img src="https://i.ibb.co/rwzkvNs/Screen-Shot-2020-02-10-at-8-35-59-PM.png" alt="Screen-Shot-2020-02-10-at-8-35-59-PM" border="0"></a>
+<a href="https://ibb.co/NN1Ycsj"><img src="https://i.ibb.co/s15382w/Screen-Shot-2020-03-30-at-6-22-04-PM.png" alt="Screen-Shot-2020-03-30-at-6-22-04-PM" border="0"></a>
 
-### mTLS self hosted
+## Sidecar to system services communication
+
+In addition to automatic mTLS between Dapr sidecars, Dapr offers mandatory mTLS between the Dapr sidecar and the Dapr system services, namely the Sentry service (Certificate Authority), Placement service (actor placement) and the Kubernetes Operator.
+
+When mTLS is enabled, Sentry writes the root and issuer certificates to a Kubernetes secret that is scoped to the namespace where the control plane is installed in. In self hosted mode, Sentry writes the certificates to a configurable filesystem path.
 
-<a href="https://ibb.co/XWFYsfY"><img src="https://i.ibb.co/rQ5d6Kd/Screen-Shot-2020-02-10-at-8-34-33-PM.png" alt="Screen-Shot-2020-02-10-at-8-34-33-PM" border="0"></a>
+In Kubernetes, when the Dapr system services start, they automatically mount the secret containing the root and issuer certs and use those to secure the gRPC server that is used by the Dapr sidecar.  
+
+In self hosted mode, each system service can be configured with the filesystem path to mount the credentials from.
```

Comment:
nit: ..each system service can be mounted to a filesystem path to get the credentials.

Reply:
addressed